strapi/strapi
# Wildcard populate performance optimizations

### What does it do?

Transform the wildcard populate (`populate=*`) to an exact populate array (`['foo', 'bar', 'baz']`) during the sanitization steps (sanitize query) to be able to run other visitors on it (remove privates, remove restricted relations, etc...).


### Why is it needed?

Prevent loading relational attributes from the database that will eventually get removed in the sanitize output step anyway due to permissions issues.

### How to test it?

Create a content type with private and non-private relational attributes
Benchmark with & without the PR.
See the improvement depending on the requested dataset's size
